src/main/java/org/ambraproject/rhino/service/impl/ArticleCrudServiceImpl.java

```diff
@@ -591,9 +591,28 @@ public Article readArticle(ArticleIdentifier articleIdentifier) {
   }
 
   @Override
-  public void updateIsPreprintOfDoi(ArticleIngestionIdentifier articleId, String preprintOfUrl) throws IOException {
+  public Collection<ArticleRevision> getPreprintArticlesWithoutVor() {
+    return hibernateTemplate.execute(session -> {
+      String subQuery = "" +
+          "SELECT DISTINCT fi.ingestion " +
+          "FROM ArticleFile fi " +
+          "INNER JOIN fi.ingestion ai " +
+          "WHERE fi.bucketName = 'preprints' " +
+          "AND ai.isPreprintOfDoi IS NULL";
+      String queryString = "" +
+          "SELECT DISTINCT ar " +
+          "FROM ArticleRevision ar " +
+          "WHERE ar.ingestion IN (" + subQuery + ")";
```

Comment:
Does HQL have an "EXISTS"?  If so, using `EXISTS` rather than `IN` may produce a faster query, and you will not need the `DISTINCT fi.ingestion`.